src/cr/cube/cube.py

```diff
@@ -380,7 +408,7 @@ def unweighted_counts(self):
     @lazyproperty
     def valid_counts(self):
         """ndarray of valid counts, valid elements only."""
-        valid_counts = self._measure.valid_counts
+        valid_counts = self._base_measure.valid_counts
         if valid_counts.any():
             return valid_counts[self._valid_idxs]
         return np.empty(0)
```

Comment:
As far as I can tell, `.valid_counts` should not be an interface method because it has no callers outside this class.

There is a broader problem though, because we shouldn't need to construct `_BaseMeasure` directly just to redundantly access a shared measure property (redundant in the sense it will be independently computed in a `_WeightedCounts` instance and perhaps an `_UnweightedCounts` as well.

What we need to do instead is this:

1. Make `_ValidCounts` a full-fledged measure of its own, that knows how to find its JSON values in the payload and shares the `.raw_cube_array` property from `_BaseMeasure` etc. `_ValidCounts` becomes the authority on what these values are and whether or not they exist in the payload; its `._flat_values` property returns `None` if the payload does not include a valid-counts measure.

2. All the logic for substituting valid-counts for unweighted and weighted counts comes out of the `._flat_values` properties of those classes and moves into `Cube.unweighted_counts` and `Cube.weighted_counts`, which become something like:

   ```python
    @lazyproperty
    def unweighted_counts(self):
        unweighted_counts = (
            self._measures.valid_counts
            if self._measures.valid_counts is not None
            else self._measures.unweighted_counts
        )
        return unweighted_counts.raw_cube_array[self._valid_idxs]
   ```

   And `Cube.weighted_counts` likewise.

3. `Cube.valid_counts_summary` uses `self._measures.valid_counts` in a similar way to get what it needs.